sdcm/nemesis.py

```diff
@@ -2102,8 +2102,7 @@ def choose_partitions_for_delete(self, partitions_amount, ks_cf, with_clustering
             if partitions_attrs.partition_range_with_data_validation:
                 exclude_partitions.extend(
                     i for i in range(partitions_attrs.partition_start_range, partitions_attrs.partition_end_range))
-                if partitions_amount == partitions_attrs.max_partitions_in_test_table:
-                    partitions_amount -= partitions_attrs.partition_end_range - partitions_attrs.partition_start_range
+                partitions_amount = min(partitions_amount, partitions_attrs.non_validated_partitions)
```

Comment:
what's the point of this ?

how would `partitions_amount` gonna bigger ?

Reply:
> what's the point of this ?
> 
> how would `partitions_amount` gonna bigger ?

If any nemesis send a bigger value for this parameter.
It is a safety measure to make sure we don't expect any unavailable number of partitions.
There can also be a chance that some partitions are deleted in the background meanwhile.